### PR TITLE
GOVC Business Logic for vSphere Setup User Command

### PR DESCRIPTION
**Pull request by jonathanmeier5**
*Issue #, if available:*
https://github.com/aws/eks-anywhere/issues/3248

*Description of changes:*
Adds govc business logic required to support setting up a vsphere user.

*Testing (if applicable):*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/executables/govc.go [hunk 1]
```diff
@@ -972,3 +972,95 @@ func getDeployOptions(network string) ([]byte, error) {
 
 	return deployOpts, err
 }
+
+func (g *Govc) CreateUser(ctx context.Context, username string, password string) error {
```
Comment (chrisdoherty4) [inline pkg/executables/govc.go:976]: We need function comments on all these even if they are simple. `AddGroupUser` for example feels a little odd. I had to look up what `sso.group.update` does which suggests the name isn't clear with its intent.

**Review by chrisdoherty4 — CHANGES_REQUESTED** [review]

File: pkg/executables/govc.go (hunk 1 above)
Comment (chrisdoherty4) [reply pkg/executables/govc.go:977]: Sorry, I should've been more elaborate. There's a specific format to Go comments too. This ensures tooling such as `go doc` can render code comments and it provides a really handy wiki for understanding the code base over the CLI. Tools like `godoc` (the web server version) also parse and publish a website similar to `pkg.go.dev`. 

https://tip.golang.org/doc/comment

Comment (chrisdoherty4) [reply pkg/executables/govc.go:977]: Heres an example that I wrote today for reference (this code may or may not get merged, eventually) https://github.com/aws/eks-anywhere/pull/3246/files

Comment (jonathanmeier5) [reply pkg/executables/govc.go:977]: Ah, thanks! I've updated the docstrings to match the style that I *think* these examples/docs are demonstrating.

**Review by chrisdoherty4 — APPROVED** [review]
lgtm

**Pull request merged**